Source/WebCore/Modules/url-pattern/URLPatternComponent.cpp

```diff
@@ -87,5 +89,40 @@ bool URLPatternComponent::matchSpecialSchemeProtocol(ScriptExecutionContext& con
     return isSchemeMatch != specialSchemeList.end();
 }
 
+JSC::JSValue URLPatternComponent::componentExec(ScriptExecutionContext& context, StringView comparedString) const
+{
+    Ref vm = context.vm();
+    JSC::JSLockHolder lock(vm);
+
+    auto regex = JSC::RegExpObject::create(vm, context.globalObject()->regExpStructure(), m_regularExpression.get(), true);
+    return regex->exec(context.globalObject(), JSC::jsString(vm, comparedString));
+}
+
+// https://urlpattern.spec.whatwg.org/#create-a-component-match-result
+URLPatternComponentResult URLPatternComponent::createComponentMatchResult(ScriptExecutionContext& context, String&& input, const JSC::JSValue& execResult) const
+{
+    URLPatternComponentResult result;
+    result.input = input;
+    URLPatternComponentResult::GroupsRecord groups;
+
+    auto globalObject = context.globalObject();
+    Ref vm = globalObject->vm();
+    auto length = execResult.get(globalObject, vm->propertyNames->length).toIntegerOrInfinity(globalObject);
+
+    unsigned index = 1;
+    while (index < length) {
+        auto match = execResult.get(globalObject, index);
+        URLPatternComponentResult::PatternMatch matchResult = !match.isNull()
+            ? URLPatternComponentResult::PatternMatch { match.toWTFString(globalObject) }
+            : URLPatternComponentResult::PatternMatch { };
+        groups.append(URLPatternComponentResult::NameMatchPair { m_groupNameList[index - 1], matchResult });
+        ++index;
+    }
+
+    result.groups = groups;
```

Comment:
`WTFMove(groups)`;
Or you should just `return URLPatternComponentResult { WTFMove(input), WTFMove(groups) };`